### DISCUSSION TITLE
Submission 2 eval, points and more

### DISCUSSION THREAD

**Discussion by fpetric** [larics/icuas24_competition#50] [Announcements]
First of all, update to the table from the first eval with the point numbers assigned to runs.

## Eval 1 results and points
| TeamID | Fruit count | Completion time | Path length | Return to start | Crashed / left area | Points |
|--------|-------------|-----------------|-------------|-----------------|---------------------|--------|
| 01     | 21          | 199.0           | 176.3       | YES             | NO                  | 43.5   |
| 02     | N/A         | N/A             | N/A         | NO              | YES                 | 0      |
| 03     | 21          | 397.3           | 204.3       | YES             | NO                  | 30     |
| 04     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 05     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 06     | 18          | 134.0           | 166.64      | YES             | YES                 | 0      |
| 07     | 18          | 285.1           | 225.6       | YES             | NO                  | 63     |
| 08     | 18          | N/A             | N/A         | NO              | YES                 | 0      |
| 09     | 23          | N/A             | N/A         | NO              | NO                  | 0      |
| 10     | 20          | 104.1           | 171.8       | YES             | YES                 | 0      |
| 11     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 12     | N/A         | N/A             | N/A         | NO              | YES                 | 0      |
| 13     | 18          | N/A             | N/A         | NO              | YES                 | 0      |
| 14     | N/A         | N/A             | N/A         | N/A             | N/A                 | 0      |
| 15     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 16     | N/A         | N/A             | N/A         | NO              | NO                  | N/A    |
| 17     | N/A         | N/A             | N/A         | NO              | NO                  | N/A    |
| 18     | N/A         | N/A             | N/A         | NO              | NO                  | N/A    |
| 19     | N/A         | N/A             | N/A         | NO              | NO                  | N/A    |
| 20     | N/A         | N/A             | N/A         | NO              | NO                  | N/A    |

Next, eval 2 results and points
## Eval 2 results and points
| TeamID | Fruit count | Completion time | Path length | Return to start | Crashed / left area | Points |
|--------|-------------|-----------------|-------------|-----------------|---------------------|--------|
| 01     | 30          | 219.3           | 178.77      | YES             | NO                  | 24.5   |
| 02     | 0           | 206.6           | 186.46      | YES             | YES                 | 0      |
| 03     | 42          | 346.0           | 193.91      | YES             | NO                  | 61.0   |
| 04     | 43          | 193.2           | 239.10      | YES             | NO                  | 57.5   |
| 05     | 41          | 136.0           | 158.49      | YES             | NO                  | 89.5   |
| 06     | 39          | 161.9           | 170.84      | YES             | YES                 | 0      |
| 07     | 41          | 236.6           | 234.95      | YES             | NO                  | 64.5   |
| 08     | 41          | N/A             | N/A         | NO              | NO                  | 0      |
| 09     | 41          | 277.5           | 175.40      | YES             | NO                  | 72.0   |
| 10     | 41          | 130.6           | 199.40      | YES             | YES                 | 0      |
| 11     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 12     | N/A         | N/A             | N/A         | NO              | YES                 | 0      |
| 13     | 41          | 112.1           | 200.54      | YES             | NO                  | 85.0   |
| 14     | N/A         | N/A             | N/A         | NO              | YES                 | 0      |
| 15     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 16     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 17     | 158         | 359.6           | 248.85      | YES             | NO                  | 8.5      |
| 18     | 41          | 140.9           | 186.22      | YES             | YES                 | 32.0    |
| 19     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |
| 20     | N/A         | N/A             | N/A         | NO              | NO                  | 0      |

## Evaluation 2 mission
Time limit was 10 minutes. The mission was "Tomato 1 3 5 7 8 13 18 21 25".  Layout was randomly generated, the plants were not the same as in the example provided in the repository. True fruit count in the assigned plant beds was 41. 

## Points calculation
Total points are a sum of total points for fruit counting (max 50 points), time taken to do the mission (max 25 points) and the path traversed to do the mission (max 25 points). The points for each category were calculated as follows:

### Fruit count
```math
points_{fruit} = 50 \cdot \left( 1 - 4 \cdot \frac{\left|count_{reported}-count_{true}\right|}{count_{true}}\right)
``` 
### Time taken
```math
points_{time} = 25 \cdot e^{\left(1-\frac{time_{team}}{time_{base}}\right)}
``` 
### Path length
```math
points_{path} = 25 \cdot e^{2\cdot\left(1-\frac{path_{team}}{path_{base}}\right)}
``` 
### Base time and path for calculation
Based on the results in the two evaluations, base time and path have been set to the following values:
```math
time_{base} = 100 s
```

```math
path_{base} = 150 m
```

### Penalties for collisions and going outside the area
For each collision or going outside the dedicated area, 25 points were deducted. 

## Reminder
Make sure to publish the fruit count to the correct topic (`/fruit_count`). Make sure to publish the fruit count only once. Make sure to return to the starting position. Team leaders will receive run logs tomorrow at the latest.

**Comment by pulak-gautam** [comment]
Would we receive feedback also on our run? (Similar to the first eval)
Thank you

**Reply by fpetric** [reply]
Help will always be given to those who ask for it :) Unfortunately, we do not have the time and resources to do individual debug sessions. But if you or anyone else has concerns, let us know via email.

**Comment by cmakelabs** [comment]
Could you please elaborate, what does "the plants were not the same as in the example provided in the repository" mean exactly

**Reply by fpetric** [reply]
The tiles and plant texture image is the same, number and layout of fruits on those tiles is not the same as in the example given in the repo. For example, in the evaluation, model tomato_1 may not have the same number of fruits as tomato_1 model in the example given in this repository. But the tile and plant image on the tile is the same. Not sure if I managed to clarify.

**Reply by fpetric** [reply]
The fruit models are the same, yes. We just moved them, removed some here, added some more there and so on.

**Reply by allenthreee** [reply]
are there some connected tomatoes or eggplants like the connected pepper as the example?

**Reply by fpetric** [reply]
There may be, but if there are some fruit that are connected, they are barely touching like in the pepper example
